### PR TITLE
Stop qualifying Type parameters with CLR

### PR DESCRIPTION
**Pull request by ajcvickers**
Part of #20409

### CODE REVIEW

**Comment by ajcvickers** [comment]
@dotnet/efcore Not sure that it is worth doing this; would like to hear what others think.

File: src/EFCore.Design/Scaffolding/Internal/TypeScaffoldingInfo.cs [hunk 1]
```diff
@@ -22,23 +22,23 @@ public class TypeScaffoldingInfo
...
+            Check.NotNull(type, nameof(type));
 
             IsInferred = inferred;
             ScaffoldUnicode = scaffoldUnicode;
             ScaffoldMaxLength = scaffoldMaxLength;
             ScaffoldPrecision = scaffoldPrecision;
             ScaffoldScale = scaffoldScale;
             ScaffoldFixedLength = scaffoldFixedLength;
-            ClrType = clrType;
+            ClrType = type;
```
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/TypeScaffoldingInfo.cs:41]: Should we call this parameter still clrType? If we had an ambiguity causing the Property to be named ClrType then it should exist for parameter too in some way.

File: src/EFCore.Relational/Metadata/IConventionSequence.cs [hunk 2]
```diff
@@ -88,10 +88,10 @@ public interface IConventionSequence : ISequence, IConventionAnnotatable
         /// <summary>
         ///     Sets the <see cref="Type" /> of values returned by the sequence.
         /// </summary>
-        /// <param name="clrType"> The <see cref="Type" /> of values returned by the sequence. </param>
+        /// <param name="type"> The <see cref="Type" /> of values returned by the sequence. </param>
         /// <param name="fromDataAnnotation"> Indicates whether the configuration was specified using a data annotation. </param>
         /// <returns> The configured value. </returns>
-        Type SetClrType([CanBeNull] Type clrType, bool fromDataAnnotation = false);
+        Type SetClrType([CanBeNull] Type type, bool fromDataAnnotation = false);
```
Comment (smitpatel) [inline src/EFCore.Relational/Metadata/IConventionSequence.cs:94]: SetClrType takes parameter named type?

File: src/EFCore/Metadata/Builders/CollectionCollectionBuilder.cs [hunk 3]
```diff
@@ -105,42 +105,42 @@ public class CollectionCollectionBuilder
...
                 {
                     ModelBuilder.RemoveImplicitJoinEntity(existingJoinEntityType);
                 }
             }
 
-            if (joinEntityType == null)
+            if (joinType == null)
             {
-                joinEntityType = ModelBuilder.Entity(joinEntityClrType, ConfigurationSource.Explicit).Metadata;
+                joinType = ModelBuilder.Entity(joinEntityType, ConfigurationSource.Explicit).Metadata;
```
Comment (smitpatel) [inline src/EFCore/Metadata/Builders/CollectionCollectionBuilder.cs:140]: This is actually decreasing readability of code. the entity type create for join table is joinType but clr Type is joinEntityType is confusing.

**Comment by smitpatel** [comment]
In few places, it feels we are just forcing something for consistency and making it ambiguous.

File: src/EFCore.Design/Scaffolding/Internal/TypeScaffoldingInfo.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore.Design/Scaffolding/Internal/TypeScaffoldingInfo.cs:41]: Or rename the property too

File: src/EFCore.Relational/Metadata/IConventionSequence.cs (hunk 2 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Metadata/IConventionSequence.cs:94]: Yes, it's already clear what it is

Comment (smitpatel) [reply src/EFCore.Relational/Metadata/IConventionSequence.cs:94]: but it is inconsistent in different dimension. :trollface:

File: src/EFCore.Design/Scaffolding/Internal/TypeScaffoldingInfo.cs (hunk 1 above)
Comment (ajcvickers) [reply src/EFCore.Design/Scaffolding/Internal/TypeScaffoldingInfo.cs:41]: Reasons I didn't rename properties:
* When using a parameter its hard to find a case where the type (Type) isn't explicitly visible in the method definition. This is not true for properties where the code may reference the property without any specific indication of type, meaning that the distinction from, for example, IEntityType, is not immediately obvious.
* It's much more breaking. We would need to obsolete all the existing properties.

File: src/EFCore.Relational/Metadata/IConventionSequence.cs (hunk 2 above)
Comment (ajcvickers) [reply src/EFCore.Relational/Metadata/IConventionSequence.cs:94]: Same reason as above for why I didn't rename methods.

File: src/EFCore/Metadata/Builders/CollectionCollectionBuilder.cs (hunk 3 above)
Comment (ajcvickers) [reply src/EFCore/Metadata/Builders/CollectionCollectionBuilder.cs:140]: Agreed.

**Comment by ajcvickers** [comment]
I don't think we're making anything ambiguous, but I also don't think this PR overall makes the code better.

**Comment by AndriySvyryd** [comment]
At least the changes to IModel and the related interfaces and extensions should still be done

**Comment by ajcvickers** [comment]
@AndriySvyryd @smitpatel So keep clrType in the type mapping code since it is pretty consistently used there and aligns with existing property names?

**Comment by smitpatel** [comment]
I am fine either way.

**Comment by AndriySvyryd** [comment]
As long as the IModel changes are in I am fine with the rest either way.

**Comment by ajcvickers** [comment]
@AndriySvyryd What about `TypeBase.ClrType`? It's in the `IModel` structure, but it's massively used and I think helps keep the consuming code clean.

**Comment by ajcvickers** [comment]
@AndriySvyryd @smitpatel New version: #22069. Leaves type mapping unchanged, but also includes some property renames/obsoletions in other places.

**Pull request closed without merging**